benches/support/macros.rs

```diff
@@ -105,22 +225,58 @@ macro_rules! bench_pathfinder {
 #[macro_export]
 macro_rules! bench_unop {
     ($b: ident, op => $unop: ident, ty => $t:ty) => {{
+        let size: usize = 1 << 13;
+        let mut rng = rand_pcg::Pcg64Mcg::new(rand::random());
+        let inputs = criterion::black_box(
+            (0..size)
+                .map(|_| <$t as mathbench::BenchValue>::random_value(&mut rng))
+                .collect::<Vec<_>>(),
+        );
+        // pre-fill output vector with some random value
+        let mut outputs = vec![<$t as mathbench::BenchValue>::random_value(&mut rng).$unop(); size];
+        let mut i = 0;
+        $b.iter(|| {
+            i = (i + 1) & (size - 1);
+            let res = unsafe { inputs.get_unchecked(i).$unop() };
+            unsafe { *outputs.get_unchecked_mut(i) = res }
+            res
+        });
+        criterion::black_box(outputs);
+    }};
+}
+
+#[macro_export]
+macro_rules! bench_unop_wide {
+    ($b: ident, width => $width: expr, op => $unop: ident, ty => $t:ty) => {{
         const SIZE: usize = 1 << 13;
+        let batch_size = (16.0 / $width as f32).ceil() as usize;
+        let total_size = SIZE * batch_size;
+
         let mut rng = rand_pcg::Pcg64Mcg::new(rand::random());
         let inputs = criterion::black_box(
-            (0..SIZE)
+            (0..total_size)
                 .map(|_| <$t as mathbench::BenchValue>::random_value(&mut rng))
                 .collect::<Vec<_>>(),
         );
         // pre-fill output vector with some random value
-        let mut outputs = vec![<$t as mathbench::BenchValue>::random_value(&mut rng).$unop(); SIZE];
+        let mut outputs = vec![<$t as mathbench::BenchValue>::random_value(&mut rng).$unop(); total_size];
         let mut i = 0;
         $b.iter(|| {
+            // minimise overhead of accessing random data using get unchecked
             i = (i + 1) & (SIZE - 1);
-            unsafe { *outputs.get_unchecked_mut(i) = inputs.get_unchecked(i).$unop() }
+            let start = i * batch_size;
+            let end = start + batch_size;
+            for j in start..end {
+                let res = unsafe { inputs.get_unchecked(j).$unop() };
+                criterion::black_box(res);
+                unsafe { *outputs.get_unchecked_mut(j) = res; }
+            }
         });
         criterion::black_box(outputs);
     }};
+    ($b: ident, op => $unop: ident, ty => $t:ty) => {{
+        bench_unop_wide!($b, width => 1, op => $unop, ty => $t)
```

Comment:
I feel like it would be a bit clearer if the width parameter always had to be explicit, so you can see from the call site what's happening. That's something I can do once merged though.